src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/DBCaseRepository.java

```diff
@@ -55,7 +54,8 @@ public DBCaseRepository(
     }
 
     public Optional<Claim> getClaimByExternalId(String externalId, String authorisation) {
-        return claimRepository.getClaimByExternalId(externalId);
+        return claimRepository.getClaimByExternalId(externalId)
+            .map(fetchedClaim -> fetchedClaim.toBuilder().id(null).build());
```

Comment:
Standard practice (in the Java community at large - I've been trying to make it so with us, too!) is to make each stream operation as simple as possible - chaining to build up the more complicated expressions. As always, judgment can be made on readability. This one would be:
```
return claimRepository.getClaimByExternalId(externalId)
    .map(Claim::toBuilder)
    .map(builder -> builder.id(null))
    .map(ClaimBuilder::build);
```